src/fluree/db/query/exec/update.cljc

```diff
@@ -138,3 +141,172 @@
 
     (retract? mdfn)
     (retract db mdfn t fuel-tracker error-ch solution-ch)))
+
+(defn retract-triple2
+  [db triple {:keys [t]} solution fuel-tracker error-ch]
+  (go
+    (try*
+      (let [retract-xf (keep (fn [f]
+                               ;;do not retract the flakes which map subject ids to iris.
+                               ;;they are an internal optimization, which downstream code
+                               ;;(eg the commit pipeline) relies on
+                               (when-not (iri-mapping? f)
+                                 (flake/flip-flake f t))))
+
+            [s-mch p-mch o-mch] (match-solution triple solution)
+
+            s-cmp  (if (string? (::where/val s-mch))
+                     (assoc s-mch ::where/val (<? (dbproto/-subid db (::where/val s-mch) true)))
+                     s-mch)
+
+            p-cmp  (if (string? (::where/val p-mch))
+                     (assoc p-mch ::where/val (<? (dbproto/-subid db (::where/val p-mch) true)))
+                     p-mch)
+            o-cmp  (if (and (= const/$xsd:anyURI (::where/val o-mch))
+                            (string? (::where/val o-mch)))
+                     (assoc o-mch ::where/val (<? (dbproto/-subid db (::where/val o-mch) true)))
+                     o-mch)]
+        (<? (where/resolve-flake-range db fuel-tracker retract-xf error-ch [s-cmp p-cmp o-cmp])))
```

Comment:
`where/resolve-flake-range` will never include an exception in the channel it returns; it will put all errors on the supplied error channel, so we should use `<!` here.

It also might include more than one chunk of flakes on the channel it returns if the flakes in question are found in more than one index node, so you have to keep taking off of the channel until it closes, and concatenate the results together.

Reply:
Will `(async/into [])` do that? It seems like it should but it's not - does the solution channel get closed when it's done getting solutions?

Reply:
`async/into` will take every thing off of a channel until it closes, and adds those items to a vector. it essentially turns a channel into a vector. this channel holds chunks of flakes, not the individual flakes themselves, so `async/into` will give you a vector that holds chunks of flakes. The solution channel does close when it's exhausted. All of the channels involved in the original update code did. 

I think this will work if you define an output channel outside of your go block, and then, instead of taking off of the channel returned by `resolve-flake-range`, you pipe it to the output channel, and then make sure to return that output channel *after* the go block:

```
(let [out-ch (async/chan)]
  (go
    ...
    (async/pipe (resolve-flake-range ...)
                out-ch))
  out-ch)
```